cmd/ocm/create/cluster/cmd.go

```diff
@@ -196,8 +206,72 @@ func init() {
 	)
 }
 
-func osdProviders() []string {
-	return []string{c.ProviderAWS, c.ProviderGCP}
+func osdProviderOptions(_ *sdk.Connection) ([]arguments.Option, error) {
+	return []arguments.Option{
+		{Value: c.ProviderAWS, Description: ""},
+		{Value: c.ProviderGCP, Description: ""},
+	}, nil
+}
+
+func getRegionOptions(connection *sdk.Connection) ([]arguments.Option, error) {
+	regions, err := provider.GetRegions(connection.ClustersMgmt().V1(), args.provider, args.ccs)
+	if err != nil {
+		return nil, err
+	}
+	options := make([]arguments.Option, len(regions))
+	for i, region := range regions {
+		// `enabled` flag only affects Red Hat infra. All regions enabled on CCS.
+		if args.ccs.Enabled || region.Enabled() {
+			options[i] = arguments.Option{
+				Value:       region.ID(),
+				Description: region.DisplayName(),
+			}
+		}
+	}
```

Comment:
@cben Wouldn't this pre-allocate a potentially larger "options" slice? if not all the regions are available as options? Perhaps `survey` knows how to deal with this but it just caught my eye.

Reply:
Oops :flushed: great catch!  Fixed.